specs/01-sharded-archiving.md

```diff
@@ -0,0 +1,288 @@
+
+#### 6. Segment Construction
+
+- Input: Buffered data of size `RAW_RECORD_SIZE`.
+- When the buffer (after encoding) contains enough data to fill a record of `RAW_RECORD_SIZE` bytes,
+  it is archived:
+  1. Split the record into `d = 2^15 =NUM_CHUNKS` chunks of size `SAFE_BYTES`.
+  2. Perform the same operations to extend with parity chunks and commit to the chunks as was done
+     for the block. This is done by creating a Merkle tree of the chunks of a record, and committing
+     to it.
+- Erasure code records column-wise with `extend(column, ERASURE_CODING_RATE)`. This effectively
+  doubles the number of rows and thus, records per segment to `NUM_PIECES` (256).
+- For each row, a proof of the row chunks is created by building a Merkle Proof of the individual
+  chunks in the share, `Cs_i`, along with their proof `ws_i`.
+- For each record, form a `piece = record || record_proof || record_proof`, `(ri∣∣Ci∣∣πi)`
+- Compute the `segment_root` by computing the Merkle proof of all the record proofs of all rows, the
+  raw records and the extended records `C_segment = Merkle_Proof(C0, C1, ..., Cn, C00, Cnn)`.
+- Append the `segment_root` to the global `segment_proofs[]` table of the chain.
+- The segment now consists of `NUM_PIECES` records of roughly 1MiB each
+  (`32 bytes * 2^15 chunks + 32 bytes proof + ~120 bytes proof`), `NUM_PIECES` piece
+  proofs, `NUM_PIECES` proofs of 32 bytes each and one 32-byte `segment_root`.
+- New pieces need to be stored by farmers and made available for retrieval by other shards and nodes
+  in the system.
+- Pieces from a shard are uniquely identified through an sequentially increasing `piece_index`.
+- Output: Segment with `NUM_PIECES` pieces and a `segment_root`.
+
+#### 7. Segment proof Submission to Beacon Chain
+
+- Input: New segments from a shard.
+- As soon as a new block is produced that includes the creation of a new segment, the segment proof
+  is submitted as a transaction in the beacon chain to commit the segment into the global history of
+  the system.
+- To submit this information to the beacon chain a `SegmentInfo` transaction is created that
+  includes the `segment_root`, the `segment_index`, and the `shard_id` of the shard where the
+  segment was created.
+- Any node in the shard (from farmers to full-nodes) are allowed to create the `SegmentInfo`
+  transaction that submits the segments into the parent's shard history.
+
+```rust
+type SegmentInfo = {
+  segment_root: Vec<u8>, // 32B
+  segment_index: u64, // 8B
+  shard_id: ShardId, // 4B
+}
+```
+
+- Output: `SegmentInfo` transaction in the beacon chain mempool.
+
+#### 8. Segment proof in beacon chain and Super Segment Construction
+
+- Input: Transaction with `SegmentInfo` in the beacon chain.
+- The execution of the `SegmentInfo` transaction when included in a block of the beacon chain
+  triggers the proof of the shard's segment in the global history of the beacon chain.
```

Comment:
"triggers" feels off here

Reply:
Rephrased it a bit (hopefully it is clearer now).